go-controller/pkg/util/pod_annotation.go

```diff
@@ -555,6 +575,11 @@ func AddRoutesGatewayIP(
 					podAnnotation.Gateways = append(podAnnotation.Gateways, gatewayIPnet.IP)
 				}
 			}
+			joinAddrs, err := ParseNodeGatewayRouterJoinAddrs(node, netinfo.GetNetworkName())
```

Comment:
this will fail on non IC, because it doesnt have the primary network in the map from what i saw
so i had to split the function and allow it to be missing 

https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4852/files#diff-01a2ef868473804cd3d9b894b093e5be0735cfb21bf27f375ece3a33cc07edc9R848-R858

Reply:
we should not fail for non IC if the IP is not there, because there is no multi path problem on non IC anyhow
of course that if it is non IC bug and it will be added then nothing need to be done
but the tests will fail now iiuc, if indeed non IC reach here (this what happened to me)

Reply:
I'd say we're pretty far behind on non-IC ...

Please open an upstream issue about this missing node annotation.

Reply:
https://github.com/ovn-kubernetes/ovn-kubernetes/issues/4876